src/main/java/com/google/swarm/tokenization/avro/ReadSplitDoFn.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.swarm.tokenization.avro;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Random;
+
+import com.google.privacy.dlp.v2.Table;
+import com.google.privacy.dlp.v2.Value;
+import org.apache.avro.Schema;
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+import org.apache.beam.sdk.io.FileIO.ReadableFile;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.values.KV;
+import org.joda.time.Instant;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Reads all records in the given split (i.e. a group of avro data blocks) and then converts
+ * those records to DLP table rows.
+ */
+public class ReadSplitDoFn extends DoFn<KV<String, ReadableFile>, KV<String, Table.Row>> {
```

Comment:
is it for avro only or will work for csv and text too? If it's only for avro may be class name should indicate that. When we move to beam transform dlp table rows created part of the transform. so we may not need this at all.

Reply:
Yes it's Avro specific. Just renamed the class. I'm not sure I follow what you mean about the transform - could you clarify?